### PR TITLE
[4] Catch, translate, and expound when GD not available.

### PR DESCRIPTION
**Pull request by PhilETaylor**
### Summary of Changes

A more user friendly error message instead of an unhandled PHP exception.

### Testing Instructions

Install Joomla 4 on a server with GD not installed or not enabled. 

Go to Joomla 4 admin -> Templates/Site Templates -> Cassiopeia Details and Files -> Click  template_thumbnail.png in left menu

### Actual result BEFORE applying this Pull Request

without debug:

Screenshot 2021-04-07 at 22 18 15

With debug:

Screenshot 2021-04-07 at 22 17 56

### Expected result AFTER applying this Pull Request

Screenshot 2021-04-07 at 22 17 37

### Documentation Changes Required

none

### CODE REVIEW

**Comment by brianteeman** [comment]
Doesnt this error also happen in media manager? It should. In which case it should be a generic mesage and not a com_template message

**Comment by PhilETaylor** [comment]
Im unable to generate a GD message in Media Manager...

**Comment by brianteeman** [comment]
I am assuming that GD is being used for the crop and rotate functionality in the template. The same functionality is present in the media manager although if memory serves me correctly the implementation is different (it shouldnt be)

**Comment by PhilETaylor** [comment]
yeah most of the image manipulation is using GD functions... Im going to rebuild a new docker container without GD and see what I can break.

**Comment by PhilETaylor** [comment]
Crop & resize certainly DO work without GD... It looks like they use `cropperjs` 1.5.11 https://fengyuanchen.github.io/cropperjs

**Comment by PhilETaylor** [comment]
This PR is ready for testing.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 417669551f67d14e7f4ee74b006bd22315e2a642<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33055">issues.joomla.org/tracker/joomla-cms/33055</a>.</sub>

File: administrator/components/com_templates/src/View/Template/HtmlView.php [hunk 1]
```diff
@@ -173,8 +173,16 @@ public function display($tpl = null)
...
-			$this->type  = 'image';
+			try
+			{
+				$this->image = $this->get('Image');
+				$this->type  = 'image';
+			}
+			catch (\RuntimeException $exception)
+			{
+				$app->enqueueMessage(Text::_('COM_TEMPLATES_GD_EXTENSION_NOT_AVAILALE'));
+				$this->type  = 'home';
```
Comment (richard67) [inline administrator/components/com_templates/src/View/Template/HtmlView.php]:
```suggestion
				$this->type = 'home';
```

**Comment by richard67** [comment]
I've restored the previous test result since the commit which invalidated the test was just a code style change.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on e7d1fbe167a52b0f9f3a3cbb3e72661929853be5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33055">issues.joomla.org/tracker/joomla-cms/33055</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33055">issues.joomla.org/tracker/joomla-cms/33055</a>.</sub>

**Comment by chmst** [comment]
Thanks!

**Pull request merged**